### PR TITLE
Allow setting of proxy and no_proxy from the constructor of LWP::UA

### LINKED ISSUE #124: Proxy attributes should be settable from the constructor
Almost all the other attributes are settable from the constructor. I cannot think of a good reason why `proxy`, `no_proxy`, and `env_proxy` should be different in this regard.

Less preferably, the documentation could be improved to explicitly mention that these are not settable from the constructor.

### PR DESCRIPTION
**Pull request by oschwald**
Closes #124

### CODE REVIEW

**Comment by oalders** [comment]
This looks good to me.  My only concern would be whether we want to introduce a dependency on `List::Util`.  It doesn't do so well on Perl 5.8 in addition to some FreeBSD failures with more modern Perls.  http://matrix.cpantesters.org/?dist=Scalar-List-Utils+1.47

**Comment by oschwald** [comment]
`List::Util` has been part of core since 5.8, although I am not sure if `pairmap` goes back that far.

**Comment by oschwald** [comment]
It looks like it was [added in 2013](https://metacpan.org/changes/distribution/Scalar-List-Utils#L155) so it isn't in 5.8. 5.8.9 has 1.19.

**Comment by oschwald** [comment]
I pushed a version that does not use `pairmap`. Feel free to squash on merge.

**Comment by oschwald** [comment]
I added the subtest plan in fbb5a5a.

File: t/base/ua.t [hunk 1]
```diff
@@ -14,6 +14,25 @@ delete $ENV{PERL_LWP_SSL_CA_FILE};
 delete $ENV{PERL_LWP_SSL_CA_PATH};
 delete $ENV{PERL_LWP_ENV_PROXY};
 
+subtest 'proxy settings from the constructor' => sub {
+    plan tests => 4;
```
Comment (genio) [inline t/base/ua.t]: That should be 3, right?

Comment (oschwald) [inline t/base/ua.t]: No, see the postfix for.

Comment (genio) [inline t/base/ua.t]: gah. right. sorry.

**Comment by oschwald** [comment]
Thanks for merging!

**Pull request merged**